### DISCUSSION TITLE
The image picker Doesn't retrieve the original file names

### DISCUSSION THREAD

**Discussion by Manikandan-saminathan** [react-native-image-picker/react-native-image-picker#2107] [Q&A]
I tried to pick the images using this library  but doesn't return the actual file name instead of it  given the generated names

**Comment by cihangir-mercan** [comment]
same :(

**Reply by cihangir-mercan** [reply]
I guess I will continue with react-native-document-picker. It returns picked file names.

**Reply by Manikandan-saminathan** [reply]
Yes But it opens the device document picker not an image gallery

**Reply by cihangir-mercan** [reply]
what is the difference since we can pick images too

**Reply by Manikandan-saminathan** [reply]
Yes! but in our current requirement we want only  the image gallery option only

**Reply by cihangir-mercan** [reply]
yeah. i returned to this too. i will use timestamps as filenames that are shown to user.

**Reply by MANIKANDAN1709** [reply]
Yes I want the exact file name only I tried to change the source code of Android but it is not working if you find any solution please update here also

**Comment by Manikandan-saminathan** [comment]
I asked the question in Twitter  also [Tweet link](https://twitter.com/college1709/status/1641786755115544577)

**Comment by cihangir-mercan** [comment]
do you know if filename comes at ios? is it only problematic at android?

**Reply by yangduonghong-duc** [reply]
same issue, only android faces this one :(
